Add sample code to use Kysely in Deno

Sleepful opened the pull request:
Hello!

It took me a while to figure out this one, here I hope to make it easier to others.

Sleepful commented on site/docs/runtimes/deno.mdx:
````diff
@@ -52,3 +52,35 @@ const sql = query.compile()
 console.log(sql.sql)
 ```
+
+Deno works well with [PostgresJS](https://github.com/porsager/postgres), to that end you can use the Kysely dialect and the driver:
````
I wasn't sure if I should remove the explanation above this one so I left it there.

Sleepful commented:
@igalklebanov hello 👋 do you think this documentation is helpful?

igalklebanov commented:
> I can't see the details of the CI failing 😅

Did you run your changes locally?

![image](https://github.com/kysely-org/kysely/assets/14938291/806f0044-70b9-4293-a5a1-bc867bcd0974)

Sleepful commented:
Oh yeah, I was using `npm start` and it didn't error.
Now that I see your screenshot, I see that I should have been running `npm run build` to check.

I updated the PR:

- Fixed some links
- Added a `/site/README.md` where I mention that `npm run build` checks for correctness. 
- Merged with `master` to reconcile the changes from your recent PR: https://github.com/kysely-org/kysely/pull/1035/files

Sleepful commented:
I can continue updating with master branch, but first I will wait to get approval or feedback on the changes

Thaigun commented:
Just letting y'all know that the example was useful fo me, in particular `DenoMigrationProvider`. 😌

Sleepful commented:
hi @igalklebanov 👋 I don't see the CI checks going beyond "waiting", would you like to add this PR? Making it easier for Deno folk could drive up Kysely adoption
